rfcs/2021-07-27-7117-vrl-string-interpolation.md

```diff
@@ -0,0 +1,188 @@
+# RFC 7117 - 2021-07-27 - VRL string interpolation
+
+VRL needs a better way to format strings. Currently the only way to do this is
+to concatenate strings, which can get unwieldly.
+
+## Scope
+
+This RFC discusses creating a new string type within VRL that can use template
+literals, also known as string interpolation, to format strings.
+
+## Pain
+
+Currently the way to create strings is through either string concatenation or
+the `join` function.
+
+Syntactically this is unwieldly. It requires extra key presses and the code
+created doesn't necessarily give an instant idea of what the resulting string
+will look like. Thus the true intent behind the code is obfuscated, which can
+result in bugs.
+
+## User Experience
+
+We will be loosely basing our format strings on Pythons [f-strings](https://peps.python.org/pep-0498/).
```

Comment:
Can you give more background on why you chose this over other alternatives? Was it random or were there specific aspects you liked?

Reply:
I've added a brief explanation:

> `f-strings` allow for a combination of embedded expressions and include the ability
to specify formatting options for the outputs. Plus, their use in a widespread language
should mean a lot of users will already be familiar with the functionality.